### DISCUSSION TITLE
Limit max number on concurrent jobs when source file is the same

### DISCUSSION THREAD

**Discussion by mpek** [RandomEngy/VidCoder#1070] [Ideas]
It would be nice if you could integrate to limit the concurrent jobs to one or a custom number, because I have set concurrent jobs to 3 and if i encode variants of the same source file it takes much longer to encode. 

The best would be to add another Option like: 

Maximum amount of concurrent jobs: "3" 
When same source file, limit maximum amount of concurrent jobs to: "1" (<- user defined custom value)

or similar. Thank you for reading and hopefully considering.

**Comment by RandomEngy** [comment]
Why would it take longer to encode 3 jobs from the same source file than 3 jobs from 3 different source files? I would think it would actually work better from the same file because you will share the disk cache hits more often.

**Comment by mpek** [comment]
Thank you for your call. I'll try to explain my observation. 
VidCoder Beta version 8.10 loop encodes the same three projects over and over again and I assumed my idea above would help solve this problem, but it seems to be a different issue that limiting jobs. 

I ran into the same problem today with two concurrent jobs. 

1. I put one job in queue and it asked for a different name and I used to auto-rename final file. 
2. Than i changed CQ value and put another job in queue, but with this job I gave the final file a different name in target box (added CQ24 to final file prior extension. 
3. Than I gave it a go to encode the two files. 
4. After a few seconds I decided to stop (context menu of jobline) the first job (the one with auto-renaming output file) and it was stopping only for a second or two and started it all over again, the second job continue to run unimpressed. 
5. I tried to stop the first job again two times, but it was only restaring it everytime I did this. 
6. After the second job reached the end, it started to encode again from the beginning, same with the first job.
7. They'll run forever, if not closing VidCoder Beta.

**Comment by RandomEngy** [comment]
I reproduced your issue and recognized another problem that could happen after manually stopping a single job. I refactored how the encode completions are tracked so it should stop endlessly looping as of 8.11 Beta.

It might still immediately restart the job if you have encode slots open and there's nothing else in the queue though.
